src/Catalog/CloudBlobStorageExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    internal static class CloudBlobStorageExtensions
+    {
+        public static async Task<IEnumerable<IListBlobItem>> ListBlobsAsync(this CloudBlobContainer container, CancellationToken cancellationToken, string prefix = null, bool useFlatBlobListing = false, BlobListingDetails blobListingDetails = BlobListingDetails.None, long? maxResults = null)
```

Comment:
Don't use optional parameters if you don't need them (or another way, keep the signature to the minimum required by the code). Hard to tell in this review if all of them are need or not.

Also note that completely forbid optional parameters on any public API, because of backcompat issues